sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -45,35 +45,28 @@ case class CatalogFunction(
  */
 case class CatalogStorageFormat(
     locationUri: Option[String],
-    inputFormat: Option[String],
-    outputFormat: Option[String],
-    serde: Option[String],
-    compressed: Boolean,
-    serdeProperties: Map[String, String]) {
+    provider: Option[String],
```

Comment:
For now we only use it to assert a table is hive table in some places, but this will be very useful when we put data source table information into catalog table instead of table properties.